c_src/erocksdb.cc

```
@@ -1235,6 +1583,45 @@ erocksdb_close(
 
 
 ERL_NIF_TERM
+erocksdb_list_column_families(ErlNifEnv* env, int argc, const ERL_NIF_TERM argv[])
+{
+    char db_name[4096];
+
+    if(!enif_get_string(env, argv[0], db_name, sizeof(db_name), ERL_NIF_LATIN1) ||
+       !enif_is_list(env, argv[1]))
+    {
+        return enif_make_badarg(env);
+    }
+
+    // parse main db options
+    rocksdb::Options *opts = new rocksdb::Options;
+    fold(env, argv[1], parse_db_option, *opts);
+
+    std::vector<std::string> column_family_names;
+
+    rocksdb::Status status = rocksdb::DB::ListColumnFamilies(*opts, db_name, &column_family_names);
+    if(!status.ok())
+        return error_tuple(env, erocksdb::ATOM_ERROR_DB_OPEN, status);
+
+    ERL_NIF_TERM result = enif_make_list(env, 0);
+    try {
+        for (size_t i = 0; i < column_family_names.size(); i++) {
+            ERL_NIF_TERM cf_name = enif_make_string(env, column_family_names[i].c_str(), ERL_NIF_LATIN1);
+            result = enif_make_list_cell(env, cf_name, result);
+        }
+    } catch (const std::exception& e) {
+        // pass through and return nullptr
```

why not return the error tuple with `e.what()` rather than `pass through`?